ChatRPG/ChatRPG/Data/FileUtility.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Text;
+
+namespace ChatRPG.Data;
+
+public class FileUtility
+{
+    private readonly string _path;
+    private readonly string _saveDir;
+    public FileUtility(string saveDir = "Saves/")
+    {
+        _saveDir = Path.Join(AppDomain.CurrentDomain.BaseDirectory, saveDir);
+        _path = SetPath(DateTime.Now);
+    }
+
+    public async Task UpdateSaveFileAsync(string message)
+    {
+        // According to .NET docs, you do not need to deck if it exists first
+        Directory.CreateDirectory(_saveDir);
+
+        await using FileStream fs =
+            new FileStream(_path, FileMode.Append, FileAccess.Write, FileShare.None, bufferSize: 4096,
+                useAsync: true);
+
+        if (!message.EndsWith("\n"))
+            message += "\n";
+
+        byte[] encodedMessage = Encoding.Unicode.GetBytes(message);
+        await fs.WriteAsync(encodedMessage, 0, encodedMessage.Length);
+    }
```

Comment:
Man kunne skrive denne metode om til at tage imod en conversation i stedet, så supporter 1..* beskeder frem for kun én, så behøver den ikke nødvendigvis at lave nye filestreams for hver besked

Reply:
Det havde jeg i starten, men så skal vi indsætte "savepoints" i koden for at sikre at conversation bliver gemt i rette tid. Hvis man f.eks. lukker programmet, så er det ikke sikkert vi får gemt det hele. 

På den her måde, så før vi appended hver message og er sikre på vi får gemt alt.

Reply:
Hmm ja okay. Jeg tænkte mest bare på om en userMessage er relevant uden dertilhørende game message, hvis vi gemmer én gang pr. par har det lidt samme behavior som en transaction (all or nothing)

Reply:
Det er sandt nok. Lige pt bliver denne metode bare kaldt to gange, når man interagere med spillet. En gang for user prompt og en gang for response. Jeg tænkte det var simpelt og selvom spillet måsket crasher, så har de stadig deres gamle prompt, hvis de godt kunne lide den. Så kan de kopiere den og prøve igen.